go/vt/vtgate/planbuilder/operators/route_planning.go

```diff
@@ -80,15 +80,19 @@ func buildVindexTableForDML(
 	ctx *plancontext.PlanningContext,
 	tableInfo semantics.TableInfo,
 	table *QueryTable,
+	ins *sqlparser.Insert,
 	dmlType string,
 ) (*vindexes.Table, Routing) {
 	vindexTable := tableInfo.GetVindexTable()
-	if vindexTable.Source != nil {
+	if tableInfo.GetVindexTable().Type == vindexes.TypeReference && vindexTable.Source != nil {
 		sourceTable, _, _, _, _, err := ctx.VSchema.FindTableOrVindex(vindexTable.Source.TableName)
 		if err != nil {
 			panic(err)
```

Comment:
The current code panics here. Is that intentional? Should we be doing something more graceful so that a bug in the parser for one query path doesn't cause vtgate to crash but let other queries through?

Reply:
Here is where we handle the panic - https://github.com/vitessio/vitess/blob/f27d287e0873d139ad2f71588068256e28a16f6a/go/vt/vtgate/planbuilder/operators/plan_query.go#L61